src/sas/readstat_xport_write.c

```diff
@@ -173,31 +224,28 @@ static readstat_error_t xport_write_variables(readstat_writer_t *writer) {
             readstat_variable_t *variable = readstat_get_variable(writer, i);
             size_t label_len = strlen(variable->label);
             size_t name_len = strlen(variable->name);
+            size_t format_len = strlen(variable->format);
             int has_long_label = 0;
             int has_long_format = 0;
-            int format_len = 0;
-            char format_name[24];
-            memset(format_name, 0, sizeof(format_name));
 
             has_long_label = (label_len > 40);
 
             if (variable->format[0]) {
-                int decimals = 2;
-                int width = 8;
-
-                int matches = sscanf(variable->format, "%s%d.%d", format_name, &width, &decimals);
-                if (matches < 1) {
-                    retval = READSTAT_ERROR_BAD_FORMAT_STRING;
-                    goto cleanup;
+                int j;
+                for (j=format_len-1; j>=0; j--) {
+                  if (!((variable->format[j] >= 48 && variable->format[j] <= 57) ||
+                          variable->format[j] == 46)) {
```

Comment:
Maybe reuse the parser from above? If not, please use the C literal syntax for characters, `'0'`, `'9'`, etc

Reply:
Yep I think best to use the parser from above once it's set up. Thanks!